src/plugins/data/public/search/expressions/esaggs/request_handler.ts

```diff
@@ -0,0 +1,213 @@
+  Object.defineProperty(requestSearchSource, 'history', {
+    get() {
+      return searchSource.history;
+    },
+    set(history) {
+      return (searchSource.history = history);
+    },
+  });
+
+  requestSearchSource.setField('aggs', function () {
+    return aggs.toDsl(metricsAtAllLevels);
+  });
+
+  requestSearchSource.onRequestStart((paramSearchSource, options) => {
+    return aggs.onSearchRequestStart(paramSearchSource, options);
+  });
+
+  // If timeFields have been specified, use the specified ones, otherwise use primary time field of index
+  // pattern if it's available.
+  const defaultTimeField = indexPattern?.getTimeField?.();
+  const defaultTimeFields = defaultTimeField ? [defaultTimeField.name] : [];
+  const allTimeFields = timeFields && timeFields.length > 0 ? timeFields : defaultTimeFields;
+
+  // If a timeRange has been specified and we had at least one timeField available, create range
+  // filters for that those time fields
+  if (timeRange && allTimeFields.length > 0) {
+    timeFilterSearchSource.setField('filter', () => {
+      return allTimeFields
+        .map((fieldName) => getTime(indexPattern, timeRange, { fieldName }))
+        .filter(isRangeFilter);
+    });
+  }
+
+  requestSearchSource.setField('filter', filters);
+  requestSearchSource.setField('query', query);
+
+  let request;
+  if (inspectorAdapters.requests) {
+    inspectorAdapters.requests.reset();
+    request = inspectorAdapters.requests.start(
+      i18n.translate('data.functions.esaggs.inspector.dataRequest.title', {
+        defaultMessage: 'Data',
+      }),
+      {
+        description: i18n.translate('data.functions.esaggs.inspector.dataRequest.description', {
+          defaultMessage:
+            'This request queries Elasticsearch to fetch the data for the visualization.',
+        }),
+        searchSessionId,
+      }
+    );
+    request.stats(getRequestInspectorStats(requestSearchSource));
+  }
+
+  try {
+    const response = await requestSearchSource.fetch({
+      abortSignal,
+      sessionId: searchSessionId,
+    });
+
+    if (request) {
+      request.stats(getResponseInspectorStats(response, searchSource)).ok({ json: response });
+    }
+
+    (searchSource as any).rawResponse = response;
+  } catch (e) {
+    // Log any error during request to the inspector
+    if (request) {
+      request.error({ json: e });
+    }
+    throw e;
+  } finally {
+    // Add the request body no matter if things went fine or not
+    if (request) {
+      request.json(await requestSearchSource.getSearchRequestBody());
+    }
+  }
+
+  // Note that rawResponse is not deeply cloned here, so downstream applications using courier
+  // must take care not to mutate it, or it could have unintended side effects, e.g. displaying
+  // response data incorrectly in the inspector.
+  let response = (searchSource as any).rawResponse;
+  for (const agg of aggs.aggs) {
+    if (typeof agg.type.postFlightRequest === 'function') {
```

Comment:
not related to this PR, but so bad :| terms other bucket only works inside expressions, not with searchsource directly